### ISSUE TITLE
One dot is missing for leading ellipsis in doc markdown

### ISSUE THREAD

**Issue by Azureblade3808** [microsoft/pylance-release#2945]
## Environment data

-   Language Server version: v2022.6.20
-   OS and version: Windows 7
-   Python version (& distribution if applicable, e.g. Anaconda): 3.8.12 Anaconda

## Code Snippet

```python
def annotation(x):
    """
    Example
    -------
    >>> class C:
    ...     @annotation
    ...     def f():
    ...         ...
    """
    return x
```

## Repro Steps

1. Copy and paste the code snippet into a python file.
2. Hover `annotation` and see its doc.

## Expected behavior

The doc markdown is rendered correctly.

## Actual behavior

Every leading ellipsis is short of one dot.
![image](https://user-images.githubusercontent.com/15389744/173978616-a7c662db-49f9-4f69-9a3d-782b90cd2bf4.png)

## Additional Info

1. It seems that a "@" phrase after leading dot(s) is causing this behavior. And a simplified sample would be

```python
def f():
    """
    >>> a
    . @1
    .. 2
    ... 3
    .... 4
    """
```

2. Wrapping the code as code blocks using \`\`\` doesn't help.

3. As far as I can tell, the same doc(s) can be rendered correctly up to v2021.1.3, but not since v2021.2.1.

**Comment by Azureblade3808** [comment]
And currectly a work-around is replacing `@` with `\x40` in docs.

**Comment by Azureblade3808** [comment]
Currently a work-around is putting a ZWSP between `@` and what's after it.

For example:
```python
def annotation(x):
    """
    Example
    -------
    >>> class C:
    ...     @\u200Bannotation
    ...     def f():
    ...         ...
    """
    return x
```

**Comment by rchiodo** [comment]
I believe this is a result of fixing cv2 doc strings in this commit:
https://github.com/microsoft/pyright/commit/e278f28dbe4dfc6f997c4e14b49a8f9eaba6317f

I'll look into if we can special case the cv2 instead.

**Comment by heejaechang** [comment]
This issue has been fixed in prerelease version 2022.10.31, which we've just released. You can find the changelog here: [CHANGELOG.md](https://github.com/microsoft/pylance-release/blob/main/CHANGELOG.md#20221031-19-October-2022-PreRelease)
